File: lib/stack.rb
# create Stack class
class Stack
    attr_reader :elements
    attr_writer :elements
    attr_reader :size
    attr_writer :size
    @elements = []
    @size = 0

    # str method
    def self.display
        if size > 0
            @elements.each do |e|
                print "#{e} "
            end
        else
            return "None"
        end
    end

    def self.size
        return @elements.size
    end

    def self.push(data)
        @elements.push(data)
    end

    def self.pop
        temp = @elements[-1]
        @elements.pop()
        return temp
    end

    def self.peek
        return @elements[-1]
     end
end

File: stack.py
'''
create a stack class that implements a stack ADT using a Python 
list to store the stack elements with index 0 as the 
bottom of the stack



'''






class Stack:
	# constructor
	def __init__(self):
		self.elements = []
		
	# make printing stack readable
	def __str__(self):
		if self.size() > 0:
			result = '| '
			for i in self.elements:
				result += str(i) + ' | '
			return result
		else:
			return 'None'

	# find size of stack
	def size(self):
		return len(self.elements)

	# add to stack
	def push(self, data):
		self.elements.append(data)

	# remove from stack
	def pop(self):
		if self.size() > 0:
			temp = self.elements[-1]
			self.elements.pop()
			return temp
		else:
			return None
	# look at top of stack
	def peek(self):
		if self.size() > 0:
			temp = self.elements[self.size() -1]
			return f'{temp}'
		else:
			return None
